### PR TITLE
Add configs for continuous integration & continuous deployment

### PR DESCRIPTION
**Pull request by bonniegee**
Updates Travis file, adds deploy scripts and configs.

I added the DJANGO_PORT as a variable available to all branches, but of course I can go with another approach.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ services:
     volumes:
       - .:/code
     ports:
-      - "8000:8000"
+      - "8000:8012"
```
Comment (helrond) [inline docker-compose.yml]:
```suggestion
      - "8012:8012"
```
Think it makes sense to just keep this all consistent

**Review by helrond — CHANGES_REQUESTED** [review]
This mostly looks good but I'd just keep the port mapping to whatever they're running on in the container. It's less complicated/easier to understand what's going on for someone coming to the app without prior knowledge.

File: deploy_scripts/substitute_env.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -e
+
+for TEMPLATE in pictor/config.py.deploy \
```
Comment (helrond) [inline deploy_scripts/substitute_env.sh]: We have the application name as an environment variable in Travis, right? So we could do 
```suggestion
for TEMPLATE in &{APPLICATION_NAME}/config.py.deploy \
```

File: deploy_scripts/stop_cron.sh.deploy [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+
+# stop all running cron jobs
+if [[ `pgrep -f pictor.*runcrons` ]]; then
+  sudo pkill -f pictor.*runcrons
```
Comment (helrond) [inline deploy_scripts/stop_cron.sh.deploy]: Same here, we could use the ${APPLICATION_NAME} variable.

**Review by helrond — CHANGES_REQUESTED** [review]
There's a couple of things I think that can be done in the deploy scripts to further abstract. Ultimately when we get around to the CD portion of our work I think we'll want to have these scripts in one place and then have apps pull them in in the deploy process, so the more we can remove hardcoded values the better.

Also, I've heard from external partners that having some comments in the config file which indicate what those configs are as well as any gotchas about values is super helpful. See Pisces for an example: https://github.com/RockefellerArchiveCenter/pisces/blob/base/pisces/config.py.example

**Review by helrond — APPROVED** [review]
Heck yeah

**Review by p-galligan — APPROVED** [review]
Looks good!

I'm assuming the secret_key in the config example is something we don't mind being public?

**Comment by bonniegee** [comment]
> I'm assuming the secret_key in the config example is something we don't mind being public?

Yep, this is not actually used in a deploy anywhere.

**Pull request merged**